### ISSUE TITLE
CI: hourly GitHub HTTP API quota is only 1000 req/hour

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#660]
After https://github.com/conbench/conbench/issues/656 and https://github.com/conbench/conbench/pull/657 we got more details about how CI may fail:

```
INFO     conbench.entities.commit:commit.py:655 got unexpected HTTP response with code 403: {"message":"API rate limit exceeded for installation ID 19416693.","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
INFO     conbench.entities.commit:commit.py:662 x-ratelimit headers: X-RateLimit-Limit: 1000, X-RateLimit-Remaining: 0, X-RateLimit-Reset: 1675259928, X-RateLimit-Used: 1001, X-RateLimit-Resource: core
INFO     conbench.tests.entities.test_commit:test_commit.py:31 exc during backfill_default_branch_commits(): Hourly GitHub HTTP API quota exhausted
```

Seen in https://github.com/conbench/conbench/actions/runs/4064840814/jobs/6998753162.

That is, the quota is only 1000 requests per hour. For regular user API tokens this [is 5000](https://docs.github.com/en/developers/apps/building-github-apps/rate-limits-for-github-apps#default-user-to-server-rate-limits-for-githubcom). Why is that only 1000 in our case here? Found the answer in https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting:

> Rate limits for requests from GitHub Actions
You can use the built-in GITHUB_TOKEN to authenticate requests in GitHub Actions workflows. For more information, see "[Automatic token authentication](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)." When using GITHUB_TOKEN, the rate limit is **1,000 requests per hour** per repository.

For now I wanted to document that finding, let's see about action items.

**Comment by sayantikabanik** [comment]
I am gonna shamelessly steal this beautiful summary for a similar (bit different) issue I faced yesterday (over `metrics` repo)

**Comment by jgehrcke** [comment]
We have addressed this via #670. Closing.
